src/core/types.ts

```diff
@@ -35,22 +35,34 @@ export type TimestampedMessage = BaseMessage & {
 };
 
 /**
- * Message that references `dataCid`.
+ * Message returned in a query result.
+ * NOTE: the message structure is a modified version of the message received, the most notable differences are:
+ * 1. does not contain `authorization`
+ * 2. may include encoded data
  */
-export type DataReferencingMessage = {
-  descriptor: {
-    dataCid: string;
-  };
-
-  encodedData: string;
+export type QueryResultEntry = {
```

Comment:
+1 to the name change from DataReferencingMessage to QueryResultEntry.

Question: Is `entry` a specific term, defined in the DWN spec or somewhere? Looks like we are using it to mean something like "the part of a message response that contains the requested data blob, plus a descriptor."

While on the subject, what is a descriptor? The DWN spec has some hyperlinks to `https://identity.foundation/decentralized-web-node/spec/#message-descriptors`, but doesn't look like that section exists (anymore?).

Reply:
> Is entry a specific term

Yes, it's defined in spec. such as https://identity.foundation/decentralized-web-node/spec/#example-response-object

> what is a descriptor

Probably easiest to think of it as a container property that "describes" the metadata of a message. The practical reason for having such a container is so that we can perform operations on the group of properties as a whole, such as computing the `recordId`.

> doesn't look like that section exists (anymore?).

Feel free to file spec issues against @csuwildcat